docs/specs/multi-registry/multi_registry.md

```diff
@@ -0,0 +1,366 @@
+
+#### Do we allow circular references between registries?
+- Short answer: No.
+- Long answer: Circular references are already prohibited inside a registry, so circular references between
+  registries can lead to the kind of problems (e.g. unresolvable references). Weaver will detect and report
+  an error if a circular reference is detected between registries.
+
+#### Do we allow references to groups or attributes defined in an imported registry?
+- Short answer: Yes.
+- Long answer: References to groups or attributes defined in an imported registry are allowed. This feature
+  allows a registry to extend or override the attributes of an imported group. Weaver will detect and report an
+  error if a reference is made to an entity that is not defined in the imported registry.
+
+### Open Questions:
+- Can we override any field of a group defined in an imported registry? No for `type` field, what about the other
+  fields?
+- Is there a relationship to define between the instrumentation scope name and version and the registry?
+
+
+## Core Policies
+
+These core policies aim to ensure consistency and backward compatibility across registries.
+
+### Group Policies
+- No group ID collisions
+- Only attribute groups are allowed in the attribute registry
+- Ref attributes are not allowed in the attribute registry
+- ID attributes are not allowed in groups outside of the attribute registry
+
+### Attribute Policies
+- No attribute name collisions
+- Attributes cannot be removed
+- Attributes cannot "degrade" in stability (e.g., stable -> experimental)
+- Stable attributes cannot change type
+- All attributes must be defined inside a semantic convention registry
+- Attribute names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+- Attributes with prefixes are not allowed; attributes must be fully qualified
+
+### Enum Member Policies
+- Stable members cannot change stability
+- Values cannot change
+- IDs cannot be removed
+- Enum IDs must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Metrics
+- No metric name collisions
+- Metrics cannot be removed
+- Metrics cannot "degrade" in stability (e.g., stable -> experimental)
+- Stable metric units cannot change
+- Stable metric instruments cannot change
+- The set of required/recommended attributes must remain the same
+- Metric names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Events
+- No event name collisions
+- Events cannot be removed
+- Events cannot "degrade" in stability (e.g., stable -> experimental)
+- The set of required/recommended attributes must remain the same
+- Event names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Spans
+- No span name collisions
+- Spans cannot be removed
+- Spans cannot "degrade" in stability (e.g., stable -> experimental)
+- The set of required/recommended attributes must remain the same
```

Comment:
I think we need to come up with more nuanced rules. E.g.
- it's ok to change recommended attribute to required. 
- it's ok to remove recommended (if you have good reasons not to populate recommended, this is your way to tell that you don't populate it) 
- it's ok to change conditionally required to recommended or required (if condition is always true)
- it might be ok to promote opt-in to recommended or even required

maybe those rules are complex enough that we just want to start with a simple one that all required attributes should stay required but some new ones may be added

Reply:
Added your comment in the spec for discussion

Reply:
+1 to starting with a simple rule (about required).

as a case in point, when the metric instrument is a monotonic counter there are very few problems associated with optional attributes: some systems will be OK, some will not. could an 3rd party library registry declare recommended attributes on a metric (say, with a configuration to enable/disable) and then an application registry pin down which attributes will or will not be used (e.g., by not enabling the choice in the 3rd party library)? that makes sense to me.